### PR TITLE
PC-9234 Replace old client and sdk with new ones

### PR DESCRIPTION
**Pull request by BSski**
Rewrote parts of the provider to
- use the new client,
- use the new SDK,

provided by the newer nobl9-go versions.

### CODE REVIEW

**Review by nieomylnieja — CHANGES_REQUESTED** [review]

File: nobl9/metadata_schema.go [hunk 1]
```diff
@@ -271,15 +225,14 @@ labelsLoop:
...
 
 	if labelsRaw != nil {
-		labelsMap := labelsRaw.(map[string]interface{})
-		for labelKey, labelValuesRaw := range labelsMap {
+		for labelKey, labelValuesRaw := range labelsRaw {
 			var labelValuesStr []string
-			for _, labelValueRaw := range labelValuesRaw.([]interface{}) {
-				labelValuesStr = append(labelValuesStr, labelValueRaw.(string))
+			for _, labelValueRaw := range labelValuesRaw {
+				labelValuesStr = append(labelValuesStr, labelValueRaw)
```
[the remark this answers is not in the document]
Comment (BSski) [reply nobl9/metadata_schema.go:235]: From what I see it also has parts related to e.g. schemas of Name, Display Name, Description and Project. Am I missing something?

[the remark this answers is not in the document]
Comment (nieomylnieja) [reply nobl9/metadata_schema.go:235]: ah damn, my bad I missed it :D

File: nobl9/resource_agent.go [hunk 2]
```diff
@@ -147,37 +148,46 @@ func resourceAgentApply(ctx context.Context, d *schema.ResourceData, meta interf
...
-	client, ds := getClient(config, project)
-	if ds.HasError() {
-		return ds
+	objects, err := client.GetObjects(ctx, project, manifest.KindAgent, nil, d.Id())
+	if err != nil {
+		return diag.FromErr(err)
 	}
 
-	objects, err := client.GetAgents("", d.Id())
+	data, err := json.Marshal(objects)
```
[the remark this answers is not in the document]
Comment (BSski) [reply nobl9/resource_agent.go:168]: Good catch, I was waiting for this one to get implemented and missed it later. Thanks!

File: examples/provider/provider.tf [hunk 3]
```diff
@@ -2,7 +2,7 @@ terraform {
   required_providers {
     nobl9 = {
       source  = "nobl9/nobl9"
-      version = "0.13.0"
+      version = "0.16.0"
```
Comment (nieomylnieja) [inline examples/provider/provider.tf:5]: please remove `organization` and `project` from the below example, let's keep it minimal:

```terraform
provider "nobl9" {
  client_id     = "<client_id>"
  client_secret = "<client_secret>"
}
```

**Review by nieomylnieja — APPROVED** [review]

**Review by nieomylnieja — DISMISSED** [review]

**Pull request merged**